daemon/limiter.go

```diff
@@ -0,0 +1,78 @@
+package daemon
+
+import (
+	"sync"
+)
+
+// limiter limits the number of concurrent goroutines running at a time by
```

Comment:
why not using a weighted semaphore: https://github.com/golang/sync/blob/master/semaphore/semaphore.go ?

Reply:
This implementation is indeed a weighted semaphore using channel coordination. I was not aware they had added a proper semaphore implementation.

I'll update this PR to use the semaphore version, as that seems more efficient.